### PR TITLE
Allow the HCL input when prompted

### PR DESCRIPTION
**Pull request by jbardin**
We already accept HCL encoded input for -vars, and this expands that to
accept HCL when prompted for a value on the command line as well.

### CODE REVIEW

File: terraform/context.go [hunk 1]
```diff
@@ -340,6 +342,12 @@ func (c *Context) Input(mode InputMode) error {
 				}
 			}
 
+			// this should only happen during tests
+			if c.uiInput == nil {
+				log.Println("[WARN] Content.uiInput is nil")
+				continue
+			}
```
Comment (phinze) [inline terraform/context.go]: lol I am sorry for the loss of sanity required to figure this bit out

File: terraform/context.go [hunk 2]
```diff
@@ -355,27 +363,33 @@ func (c *Context) Input(mode InputMode) error {
...
-					break
+					if v.Required() {
+						// Redo if it is required, but abort if we keep getting
+						// blank entries
+						if retry > 2 {
+							return fmt.Errorf("missing required value for %q", n)
+						}
+						retry++
+						continue
+					}
```
Comment (phinze) [inline terraform/context.go]: Any reason this got fanned out from `&&` before to nested `if` here?

**Comment by phinze** [comment]
LGTM w/ one tiny Q

File: terraform/context.go (hunk 2 above)
Comment (jbardin) [inline terraform/context.go]: I think there was a statement after the nested if during some iteration. I'll collapse this back down again before merging.

**Pull request merged**